apiserver/crossmodel/crossmodel.go

```diff
@@ -23,37 +25,35 @@ func init() {
 // API implements the cross model interface and is the concrete
 // implementation of the api end point.
 type API struct {
-	authorizer        facade.Authorizer
-	applicationOffers jujucrossmodel.ApplicationOffers
-	backend           Backend
+	authorizer           facade.Authorizer
+	getApplicationOffers func(interface{}) jujucrossmodel.ApplicationOffers
+	backend              Backend
+	statePool            StatePool
 }
 
 // createAPI returns a new cross model API facade.
 func createAPI(
-	applicationOffers jujucrossmodel.ApplicationOffers,
+	getApplicationOffers func(interface{}) jujucrossmodel.ApplicationOffers,
```

Comment:
You can't pass a specific type because *state.State and *stateShim are not identical and you get a compile error.